src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/SurfaceNets.cpp

```diff
@@ -0,0 +1,388 @@
+
+  // Export an OBJ file for each material to the specified path
+  std::vector<int> materials = geometry->labels();
+  for(const auto& itMatIdx : materials)
+  // for(std::vector<int>::iterator itMatIdx = materials.begin(); itMatIdx != materials.end(); itMatIdx++)
+  {
+    if(itMatIdx > 20 && itMatIdx < 65530)
+    {
+      continue;
+    }
+    std::string filename = fmt::format("/tmp/surface_mash_test_{}.obj", itMatIdx);
+    std::cout << "Export file Obj file: " << filename << std::endl;
+    std::ofstream stream(filename, std::ios_base::binary);
+    if(stream.is_open())
+    {
+      MMGeometryOBJ::OBJData data = geometry->objData(itMatIdx);
+      stream << "# vertices for feature id " << itMatIdx << std::endl;
+      // for(std::vector<std::array<float, 3>>::iterator v = data.vertexPositions.begin(); v != data.vertexPositions.end(); v++)
+      for(const auto& vertex : data.vertexPositions)
+      {
+        stream << "v " << (vertex)[0] << ' ' << (vertex)[1] << ' ' << (vertex)[2] << std::endl;
+      }
+      // for(std::vector<std::array<int, 3>>::iterator t = data.triangles.begin(); t != data.triangles.end(); t++)
+      stream << "# triangles for feature id " << itMatIdx << std::endl;
+      for(const auto& face : data.triangles)
+      {
+        stream << "f " << (face)[0] << ' ' << (face)[1] << ' ' << (face)[2] << std::endl;
+      }
+    }
+  }
+}
+
+} // namespace
+// -----------------------------------------------------------------------------
+SurfaceNets::SurfaceNets(DataStructure& dataStructure, const IFilter::MessageHandler& mesgHandler, const std::atomic_bool& shouldCancel, SurfaceNetsInputValues* inputValues)
+: m_DataStructure(dataStructure)
+, m_InputValues(inputValues)
+, m_ShouldCancel(shouldCancel)
+, m_MessageHandler(mesgHandler)
+{
+}
+
+// -----------------------------------------------------------------------------
+SurfaceNets::~SurfaceNets() noexcept = default;
+
+// -----------------------------------------------------------------------------
+const std::atomic_bool& SurfaceNets::getCancel()
+{
+  return m_ShouldCancel;
+}
+
+// -----------------------------------------------------------------------------
+Result<> SurfaceNets::operator()()
+{
+  // Get the ImageGeometry
+  auto& imageGeom = m_DataStructure.getDataRefAs<ImageGeom>(m_InputValues->GridGeomDataPath);
+
+  // Get the Created Triangle Geometry
+  TriangleGeom& triangleGeom = m_DataStructure.getDataRefAs<TriangleGeom>(m_InputValues->TriangleGeometryPath);
+
+  auto gridDimensions = imageGeom.getDimensions();
+  auto voxelSize = imageGeom.getSpacing();
+  IntVec3 arraySize(static_cast<int32>(gridDimensions[0]), static_cast<int32>(gridDimensions[1]), static_cast<int32>(gridDimensions[2]));
+
+  Int32Array& featureIds = m_DataStructure.getDataRefAs<Int32Array>(m_InputValues->FeatureIdsArrayPath);
+  auto& featureIdDataStore = featureIds.getIDataStoreRefAs<Int32DataStore>();
```

Comment:
```suggestion
  auto& featureIdDataStore = featureIds.getDataStoreRef();
```
Using `getDataStoreRef()` returns an `AbstractDataStore<T>&` that is expandable to other forms of data storage. If you do not require `DataStore<T>`'s `data()` method, there is no need to force the use of an in-memory implementation.